.github/actions/templates/validateModulePSRule/action.yml

```diff
@@ -0,0 +1,131 @@
+            tenantId          = '${{ env.ARM_TENANT_ID }}'
+          }
+
+          ## Add local (source control) tokens
+          $tokenMap = @{}
+          foreach ($token in (Get-ChildItem env: | Where-Object -Property Name -Like "localToken_*")) {
+            $tokenMap += @{ $token.Name.Replace('localToken_','','OrdinalIgnoreCase') = $token.value }
+          }
+          Write-Verbose ('Using local tokens [{0}]' -f ($tokenMap.Keys -join ', ')) -Verbose
+          $Tokens += $tokenMap
+
+          ## Swap 'namePrefix' token if empty and provided as a GitHub secret
+          if([String]::IsNullOrEmpty($Tokens['namePrefix'])){
+            Write-Verbose 'Using [namePrefix] token from GitHub' -Verbose
+            $Tokens['namePrefix'] = '${{ env.TOKEN_NAMEPREFIX }}'
+          }
+
+          # Construct Token Function Input
+          $ConvertTokensInputs = @{
+            FilePathList = @($templateFilePath)
+            Tokens       = $Tokens
+            TokenPrefix  = '${{ env.tokenPrefix }}'
+            TokenSuffix  = '${{ env.tokenSuffix }}'
+          }
+
+          Write-Verbose "Convert Tokens Input:`n $($ConvertTokensInputs | ConvertTo-Json -Depth 10)" -Verbose
+
+          # Invoke Token Replacement Functionality [For Module]
+          $null = Convert-TokensInFileList @ConvertTokensInputs
+
+          Write-Output '::endgroup::'
+
+    # Run analysis by using the PSRule GitHub action.
+    - name: Run PSRule analysis
+      uses: microsoft/ps-rule@v2.4.0
+      continue-on-error: true # Setting this whilst PSRule gets bedded in, in this project
+      with:
+        modules: 'PSRule.Rules.Azure'
+        inputPath: '${{ inputs.templateFilePath}}'
+        outputFormat: Csv
+        outputPath: '${{ inputs.templateFilePath}}-PSRule-output.csv'
+
+    - name: 'Parse CSV content'
+      if: always()
+      uses: azure/powershell@v1
+      with:
+        azPSVersion: 'latest'
+        inlineScript: |
+          # Grouping task logs
+          Write-Output '::group::Parse CSV content'
+
+          # Load used functions
+          . (Join-Path $env:GITHUB_WORKSPACE 'utilities' 'pipelines' 'PSRuleValidation' 'Set-PSRuleGitHubOutput.ps1')
+
+          # Populate parameter input
+          $ParameterInput = @{
+            inputFilePath           = '${{ inputs.templateFilePath}}-PSRule-output.csv'
+            outputFilePath          = '${{ inputs.templateFilePath}}-PSRule-output.md'
+            skipPassedRulesReport   = $false
+          }
+
+          Write-Verbose "Set PS Rule Output with following parameters:`n $($Input | ConvertTo-Json -Depth 10)" -Verbose
+
+          # Invoke Set PSRule Output Functionality
+          $null = Set-PSRuleGitHubOutput @ParameterInput
+
+          Write-Output '::endgroup::'
+
+    - name: Output to GitHub job summaries
+      if: always()
+      shell: pwsh
+      run: Get-Content '${{ inputs.templateFilePath}}-PSRule-output.md' >> $env:GITHUB_STEP_SUMMARY
```

Comment:
If the previous step fails (to create the file) - won't this fail too? Maybe it would be worth to add a simple error handling via `Test-Path` that provides a meaningful error message.

Reply:
Added check to both inline script and utility used by the previous task